File: temp/ComputerStore/library/src/Goods/PC.cpp
#include "Goods/PC.h"
#include "Exception/GoodException.h"

PC::PC(double price, int ram, int ssdCapacity, processor CPU, blockType block) : Computer(price, ram, ssdCapacity,
                                                                                          CPU),block(block) {
    if(ram <=256) throw GoodExeption("ram is less then 256 MB");
}

PC::~PC() {}

blockType PC::getBlockType() const {
    return block;
}

double PC::getPrice() {
    return Computer::getPrice() + (block+1)*500;
}

std::string PC::toString() {
    return Computer::toString() +
           "\nBlock Type: " + std::to_string(getBlockType()) + // problem is possible (enum)
           "\nPrice: " + std::to_string(getPrice());
}

File: src/main/java/model/goods/PC.java
package model.goods;

import controller.exceptions.GoodException;
import lombok.ToString;

@ToString(callSuper = true)
public class PC extends Computer {
    public PC() {
        super();
    }

    public PC(double basePrice,String goodName, int ram, int ssdCapacity) throws GoodException {
        super(basePrice, goodName, ram, ssdCapacity);
    }
}
